logger.go

```diff
@@ -121,7 +142,7 @@ func WithLogger(ctx context.Context, logger *Logger) context.Context {
 
 func FromContext(ctx context.Context) *Logger {
 	if logger, ok := ctx.Value(loggerKey{}).(*Logger); ok {
-		return logger
+		return &Logger{Logger: logger.Logger, ctx: ctx}
```

Comment:
Does this alloc on every `FromContext`?

Reply:
It does now, yes. Do we want to profile it to see if it's a problem?